src/main/java/uk/gov/hmcts/cmc/claimstore/events/EventProducer.java

```diff
@@ -27,22 +31,31 @@
 @Component
 public class EventProducer {
     private final ApplicationEventPublisher publisher;
+    private final boolean asyncEventOperationEnabled;
 
-    public EventProducer(ApplicationEventPublisher publisher) {
+    @Autowired
+    public EventProducer(
+        ApplicationEventPublisher publisher,
+        @Value("feature_toggles.async_event_operations_enabled") String asyncEventOperationEnabled
+    ) {
         this.publisher = publisher;
+        this.asyncEventOperationEnabled = Boolean.getBoolean(asyncEventOperationEnabled);
     }
 
-    public void createClaimIssuedEvent(
-        Claim claim,
-        String pin,
-        String submitterName,
-        String authorisation
-    ) {
+    public void createClaimIssuedEvent(Claim claim, String pin, String submitterName, String authorisation) {
 
-        if (claim.getClaimData().isClaimantRepresented()) {
-            publisher.publishEvent(new RepresentedClaimIssuedEvent(claim, submitterName, authorisation));
+        if (asyncEventOperationEnabled) {
```

Comment:
Please split this method in two, one for `ClaimIssuedEven` and other for `ClaimCreatedEvent`